lib/codeowners/cleaner.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'git'
+require 'logger'
+
+require_relative 'checker/code_owners'
+require_relative 'checker/file_as_array'
+require_relative 'checker/whitelist'
+
+module Codeowners
+  # Cleans a codeowner file by sorting, grouping and removing
+  # useless patterns, all automatically
+  class Cleaner
+    def initialize(repo)
+      @git = Git.open(repo, log: Logger.new(IO::NULL))
+      @filename = Checker::CodeOwners.filename(repo)
+    end
+
+    def clean!
```

Comment:
Please add high-level documentation to the public method.
We can mention the steps taken into account.
- Remove duplicated lines
- Simplify patterns
- Remove useless pattern

Reply:
Done